Sources/PackageRegistryTool/PackageRegistryTool+ArchiveSign.swift

```diff
@@ -0,0 +1,469 @@
+            observabilityScope.emit(info: "signing \($0)")
+            let signedManifestPath = workingDirectory.appending($0)
+
+            var manifest = try fileSystem.readFileContents(packageDirectory.appending($0)).contents
+            let signature = try SignatureProvider.sign(
+                content: manifest,
+                identity: signingIdentity,
+                intermediateCertificates: intermediateCertificates,
+                format: signatureFormat,
+                observabilityScope: observabilityScope
+            )
+            manifest
+                .append(
+                    contentsOf: Array(
+                        "\n// signature: \(signatureFormat.rawValue);\(Data(signature).base64EncodedString())"
+                            .utf8
+                    )
+                )
+
+            try fileSystem.writeFileContents(signedManifestPath) { stream in
+                stream.write(manifest)
+            }
+        }
+
+        // create the archive
+        observabilityScope.emit(info: "archiving the source at '\(packageDirectory)'")
+        let archivePath = try PackageArchiver.archive(
+            packageIdentity: packageIdentity,
+            packageVersion: packageVersion,
+            packageDirectory: packageDirectory,
+            workingDirectory: workingDirectory,
+            workingFilesToCopy: manifests,
+            cancellator: cancellator,
+            observabilityScope: observabilityScope
+        )
+        let archive = try localFileSystem.readFileContents(archivePath).contents
+
+        // sign the archive
+        observabilityScope.emit(info: "signing the archive at '\(archivePath)'")
+        let archiveSignature = try SignatureProvider.sign(
+            content: archive,
+            identity: signingIdentity,
+            intermediateCertificates: intermediateCertificates,
+            format: signatureFormat,
+            observabilityScope: observabilityScope
+        )
+        let archiveSignaturePath = workingDirectory.appending("\(packageIdentity)-\(packageVersion).sig")
+        try fileSystem.writeFileContents(archiveSignaturePath) { stream in
+            stream.write(archiveSignature)
+        }
+
+        var signedMetadata: SignedItem? = .none
+        if let metadataPath {
+            observabilityScope.emit(info: "signing metadata at '\(metadataPath)'")
+            let metadata = try localFileSystem.readFileContents(metadataPath).contents
+            let metadataSignature = try SignatureProvider.sign(
+                content: metadata,
+                identity: signingIdentity,
+                intermediateCertificates: intermediateCertificates,
+                format: signatureFormat,
+                observabilityScope: observabilityScope
+            )
+            let metadataSignaturePath = workingDirectory.appending("\(packageIdentity)-\(packageVersion)-metadata.sig")
+            try fileSystem.writeFileContents(metadataSignaturePath) { stream in
```

Comment:
I think we have a help for writeFileContents that does not require going throw the steam